lxd/network/openvswitch/ovn.go

```diff
@@ -1880,14 +1880,49 @@ func (o *OVN) PortGroupPortClearACLRules(portGroupName OVNPortGroup, portName OV
 	return nil
 }
 
+// loadBalancerUUIDs returns list of UUID records for named load balancer.
+func (o *OVN) loadBalancerUUIDs(loadBalancerName OVNLoadBalancer) ([]string, error) {
+	lbTCPName := fmt.Sprintf("%s-tcp", loadBalancerName)
+	lbUDPName := fmt.Sprintf("%s-udp", loadBalancerName)
+
+	var lbUUIDs []string
+
+	// Use find command in order to workaround OVN bug where duplicate records of same name can exist.
+	for _, lbName := range []string{lbTCPName, lbUDPName} {
+		output, err := o.nbctl("--format=csv", "--no-headings", "--data=bare", "--colum=_uuid", "find", "load_balancer",
+			fmt.Sprintf(`name="%s"`, lbName),
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		lbUUIDs = append(lbUUIDs, shared.SplitNTrimSpace(strings.TrimSpace(output), "\n", -1, true)...)
+	}
+
+	return lbUUIDs, nil
+}
+
 // LoadBalancerApply creates a new load balancer (if doesn't exist) on the specified routers and switches.
 // Providing an empty set of vips will delete the load balancer.
 func (o *OVN) LoadBalancerApply(loadBalancerName OVNLoadBalancer, routers []OVNRouter, switches []OVNSwitch, vips ...OVNLoadBalancerVIP) error {
 	lbTCPName := fmt.Sprintf("%s-tcp", loadBalancerName)
 	lbUDPName := fmt.Sprintf("%s-udp", loadBalancerName)
 
-	// Remove existing load balancers if they exist.
-	args := []string{"--if-exists", "lb-del", lbTCPName, "--", "lb-del", lbUDPName}
+	// Remove load balancers if they exist.
+	lbUUIDs, err := o.loadBalancerUUIDs(loadBalancerName)
+	if err != nil {
+		return fmt.Errorf("Failed getting UUIDs: %w", err)
```

Comment:
Would it be worth providing also the name of the LB in the error message? Something like 
`Failed getting UUIDs for load balancer %q: %w`?

Reply:
I was in two minds about this - the load balancer underlying concept is used for both LXD forwarders and load balancers, so I refrained on purpose from potentially exposing an error about the load balancers when it maybe in the context of a network forward. Instead I went for adding some context just about what phase of the apply process was failing.

Reply:
That makes total sense, fine for me then.